presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -732,7 +742,13 @@ protected Type visitArithmeticUnary(ArithmeticUnaryExpression node, StackableAst
         @Override
         protected Type visitArithmeticBinary(ArithmeticBinaryExpression node, StackableAstVisitorContext<Context> context)
         {
-            return getOperator(context, node, OperatorType.valueOf(node.getOperator().name()), node.getLeft(), node.getRight());
+            Type returnType = getOperator(context, node, OperatorType.valueOf(node.getOperator().name()), node.getLeft(), node.getRight());
+            if (sqlFunctionProperties.shouldWarnOnPossibleNans() && node.getOperator() == ArithmeticBinaryExpression.Operator.DIVIDE && TypeUtils.isApproximateNumericType(returnType)) {
```

Comment:
This isn't the only way to get NaN. You can also get it from a sqrt or power function, mulitplying Inf by 0, and probably other ways I'm not thinking of.

Reply:
yeah, there are definitely other ways of producing nans. Maybe I'll call this warn_on_common_nan_patterns?